SpatialGDK/Source/SpatialGDK/Public/LoadBalancing/LayeredLBStrategy.h

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "LoadBalancing/AbstractLBStrategy.h"
+
+#include "CoreMinimal.h"
+#include "Math/Box2D.h"
+#include "Math/Vector2D.h"
+
+#include "LayeredLBStrategy.generated.h"
+
+class SpatialVirtualWorkerTranslator;
+class UAbstractLockingPolicy;
+
+DECLARE_LOG_CATEGORY_EXTERN(LogLayeredLBStrategy, Log, All)
+
+USTRUCT()
+struct FLBLayerInfo
+{
+	GENERATED_BODY()
+
+	FLBLayerInfo() : Name(NAME_None)
+	{
+	}
+
+	UPROPERTY()
+	FName Name;
+
+	UPROPERTY(EditAnywhere, Category = "Load Balancing")
+	TSubclassOf<UAbstractLBStrategy> LoadBalanceStrategy;
+
+	UPROPERTY(EditAnywhere, Category = "Load Balancing")
+	TSubclassOf<UAbstractLockingPolicy> LockingPolicy;
+};
+
+/**
+ * A load balancing strategy that wraps multiple LBStrategies. The user can define "Layers" of work, which are
+ * specified by sets of classes, and a LBStrategy for each Layer. This class will then allocate virtual workers
+ * to each Layer/Strategy and keep track of which Actors belong in which layer and should be load balanced
+ * by the corresponding Strategy.
+ */
+UCLASS()
+class SPATIALGDK_API ULayeredLBStrategy : public UAbstractLBStrategy
+{
+	GENERATED_BODY()
+
+public:
+	ULayeredLBStrategy();
+	~ULayeredLBStrategy();
+
+	/* UAbstractLBStrategy Interface */
+	virtual void Init() override;
+
+	virtual void SetLocalVirtualWorkerId(VirtualWorkerId InLocalVirtualWorkerId) override;
+
+	virtual bool ShouldHaveAuthority(const AActor& Actor) const override;
+	virtual VirtualWorkerId WhoShouldHaveAuthority(const AActor& Actor) const override;
+
+	virtual SpatialGDK::QueryConstraint GetWorkerInterestQueryConstraint() const override;
+
+	virtual bool RequiresHandoverData() const { return GetMinimumRequiredWorkers() > 1; }
```

Comment:
I guess we should put override in here for consistency? Or is it ignored if you define the method inline?
```suggestion
	virtual bool RequiresHandoverData() const override { return GetMinimumRequiredWorkers() > 1; }
```